framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -517,18 +544,40 @@ protected boolean visitTypeArgs(
             : AnnotationUtils.getElementValueArray(
                 covariantAnno, covariantValueElement, Integer.class);
 
-    for (int i = 0; i < supertypeTypeArgs.size(); i++) {
-      final AnnotatedTypeMirror superTypeArg = supertypeTypeArgs.get(i);
-      final AnnotatedTypeMirror subTypeArg = subtypeTypeArgs.get(i);
-      final boolean covariant = covariantArgIndexes != null && covariantArgIndexes.contains(i);
-
-      boolean result = isContainedBy(subTypeArg, superTypeArg, covariant);
+    // JLS: 4.10.2. Subtyping among Class and Interface Types
+    // 4th paragraph
```

Comment:
I think this is only handling the 3rd bullet point of that paragraph.  Does it actually handle more?

Nit: I would change "JLS" to "JLS 11" in case later editions rewrite the section.

Reply:
I answered this in this [comment](https://github.com/typetools/checker-framework/pull/3922/files/8d2e1d98be7e04c8cae2d2da14bc6029af7ab759#r661831678).